Pylint, ja CI

ToxPuro opened the pull request.

ToxPuro committed:
tests run on pull request

ToxPuro committed:
pylint on CI and refactored code to match what pylint wanted

ToxPuro committed:
fixed typo

ToxPuro committed:
pylint passes

ollpu commented on src/routes.py:
```diff
@@ -22,16 +24,14 @@ def userpage():
-        else:
-            flash(f"Login failed")
-            return redirect("/")
+        flash("Login failed")
+        return redirect("/")
```
I dislike this specific lint (`no-else-return`), because it goes against common practices and does not express intent very well. Either way, no big deal in this project.

ToxPuro replied:
I think you are right that with else the intent can be expressed better

ollpu commented on .github/workflows/cd.yml:
```diff
@@ -20,13 +23,18 @@ jobs:
         run: pip install poetry
       - name: Install dependencies
         run: poetry install
+      - name: Run pylint
+        run: poetry run pylint src
```
This fails the whole job if there is a single warning. While we should strive for a perfect score, and warnings can be suppressed if really necessary, it might be too limiting.

Consider adding e.g. `--fail-under=8`

ToxPuro replied:
added

ollpu commented on .github/workflows/cd.yml:
```diff
@@ -20,13 +23,18 @@ jobs:
         run: poetry install
+      - name: Run pylint
+        run: poetry run pylint src
       - name: Run tests
         run: poetry run coverage run -m pytest
```
We could add a check that coverage is sufficient after this:

```sh
poetry run coverage report --fail-under=80
```

(80% was recommended by Kalle, though the current DoD says 70%)

ToxPuro replied:
Good idea!

ollpu reviewed (changes requested):
Looks good overall, aside from the  `--fail-under` additions I have mentioned in comments.

Can you also set up running Robot tests in CI or would that be a separate PR?

ToxPuro committed:
robot tests in CI

ToxPuro committed:
fixed missing &

ToxPuro commented:
added robot tests to CI in this branch

ollpu commented:
> added robot tests to CI in this branch

Great, though it seems that the tests are unable to run, but the workflow still passes. Something to do with there being no display / headless mode?

ToxPuro committed:
Robot tests run with headlesschrome in CI

ToxPuro committed:
trying to add delay for the headless chrome

ToxPuro commented:
I changed that tests are run in headless mode at CI. There seems to be a bug where headless chrome doesn't open the window large enough that the links aren't in the page

ToxPuro committed:
changed Maximize Browser Window with Set Window Size, since with headlesschrome tests didn't run

ToxPuro committed:
change slow delay value to something more suitable

ToxPuro commented:
Now Robots tests run properly

ollpu committed:
Pass through return value of robot tests

The "trap" command should be executed before the tests start,
so interrupts perform correct cleanup. Also kill pid gotten from "jobs -p"
instead of "lsof", this should avoid accidentally killing outside
processes.

The last statement is now the actual robot command, so its return value
is passed through to CI correctly.

ollpu commented:
I changed the `run_robot_tests.sh` script somewhat, so that the exit value is passed through to CI correctly. Should be ready to merge now.

ToxPuro committed:
Testing that step fails in actions if tests fail

ToxPuro committed:
Actions works now

ToxPuro commented:
Great, now CI works correctly
